kernel/exec.c

```diff
@@ -1094,27 +1094,24 @@ int myst_exec(
     free(hashbang_buff);
     hashbang_buff = NULL;
 
-    /* if this is a nested exec, then release previous exec's stack */
-    if (process->exec_stack)
-    {
-        myst_munmap(process->exec_stack, process->exec_stack_size);
-        process->exec_stack = NULL;
-        process->exec_stack_size = 0;
-    }
+    /* if this is a nested exec, then release previous process mappings */
+    myst_release_process_mappings(process->pid);
```

Comment:
For all process threads except the main thread, we release process mappings when the thread long jumps back to the exit sequence [in _run_thread](https://github.com/deislabs/mystikos/blob/main/kernel/thread.c#L1054). Is there a reason to duplicate it/move it here?

Reply:
Oh nevermind, the preceding comment makes sense now. However is there a check for nested exec missing?

Reply:
This call covers the nested exec already:
- For non-nested exec, it acts as no-op as there is no mapping associated with the process->pid
- For nested exec, it releases the previous mapping associated with the process->pid (retaining the exiting behavior).

Reply:
> For non-nested exec, it acts as no-op as there is no mapping associated with the process->pid

How about crt_data and exec_stack?

Reply:
This PR adds these two region to the pid vectors, which will be freed by myst_release_process_mappings per pid

Reply:
I'm still a bit confused. For the non-nested exec case, crt_data and exec_stack are being `myst_mmap`ed earlier in the function. Wouldn't an unconditional release of mappings here cause issues?

Reply:
The pid vectors update is done after here, see line 1101 - 1106. The call here only affects the existing mapping associated with the pid. The newly allocated stack/crt are set to the `process` later.

Reply:
Ah gotcha, thanks for clarifying that. Could you elaborate on this in the code comment for future reference of someone looking at this code.